app/main/views/frameworks.py

```diff
@@ -212,32 +212,40 @@ def get_framework_suppliers(framework_slug):
         Framework.slug == framework_slug
     ).first_or_404()
 
-    # TODO: Work out a way to order by current_framework_agreement.signed_agreement_returned_at
-    # Listing agreements is something done for Admin only (suppliers only retrieve their individual agreements)
-    # CCS always want to work from the oldest returned date to newest, so order by ascending date
+    # we're going to need an alias for the current_framework_agreement join
```

Comment:
We don't actually need to tell it to `joinedload` - specifying the relationship on the model with `lazy="joined"` will make it do that by default. What we _need_ to do is reuse this join for a filter and/or an ordering. To do that we need to grab a handle on it to be able to refer to it in `filter` and `order_by` clauses.

The _effect_ of this collection of lines is to bind the python-alias `cfa` to that existing join.

What `contains_eager` _literally_ does is say - "this query here - this already _has_ an equivalent join to what `.current_framework_agreement` asks for, and it's in the join we've 'aliased' to `cfa` (using the previous `.outerjoin` call), so.. please reuse it instead of creating yet another join to satisfy what `.current_framework_agreement` wants".